pyproject.toml

```diff
@@ -282,10 +296,15 @@ omit = [
 # The test suite spawns subprocesses to test some stuff, so make sure
 # this doesn't corrupt the coverage files
 parallel = true
+plugins = []
+relative_files = true
+source = ["."]
```

Comment:
Why do we need source and source_pkgs? We should only need one

Reply:
This is needed to measure coverage outside `src/`. One setting can only contain importable names while the other can have both importables and paths.

I had to do a lot of experiments to arrive to this config shape. Other variants may output incorrect paths into the XML report, confusing Codecov.

Reply:
(this also allows including that Cython smoke test)

Reply:
Probably that smoke test should be inside src/trio/_test

Reply:
Alternatively we could use `source_pkgs = ["trio", "tests"]`

Reply:
@graingert this won't work well. The combination of the two is the only config that is known not to cause problems one way or the other. Plus, it represents the shape of the project exactly — the source is the entire Git repo; the importable is whatever is in `site-packages` but it's also mapped back to the source for proper representation within both coveragepy and codecov.

I've seen cases where it would display `test_*.py` files in the root of the repository in Codecov while showing the correct paths in the coveragepy's own HTML output. And that's just one example of how things tend to break.

Can we please rely on my experience here? I happened to maintain the use of coveragepy specifically a lot in the past few years, and would like to avoid weird behaviors just because this project wants to deviate from what's gone through countless hours of debugging rather surprising corner cases.

Reply:
I added the source_pkgs feature to coveragepy for this use case

Reply:
@graingert really? I didn't know!

Still, there were cases where this specific configuration layout was the only thing that made it possible for Codecov not to break in some cases. This was in both src- and flat-layout projects (some in aio-libs, some elsewhere).

Though, I feel like it's still important to cover other things that might appear outside `src/`. Sometimes, these are helper scripts or something similar — by collecting such coverage, it is possible to configure Codecov-reported statuses better and be more conscious about what to take into account through exclusions.

As for adding `"tests"` into `source_pkgs`, that might work, but would not address the “covering everything” part. Plus, I'm worried about the side effects it may have on Codecov.